### ISSUE TITLE
cannot install with python3. 

### ISSUE THREAD

**Issue by PharrellWANG** [ray-project/ray#560]
<img width="967" alt="screen shot 2017-05-17 at 2 56 56 pm" src="https://cloud.githubusercontent.com/assets/22740944/26141976/16d11fdc-3b11-11e7-8d5b-ced911f48e93.png">
cannot select correct python version. sticking to python2 when installing ray.

**Comment by robertnishihara** [comment]
This happens when you use Ray with Python 3, but during compilation it accidentally compiled against the Python 2 libraries.

I've seen this happen for a couple of reasons.

1. You previously installed it with Python 2, CMake found Python 2 and cached the version of Python it found. You then tried installing with Python 3 and CMake didn't look for different Python libraries.

2. You installed Ray with a command other than `python setup.py install` (maybe instead of `python` you typed `Python` or `python3`). Ray currently searches for the correct Python libraries to compile against by using `which python`, so if `which python` doesn't find the right Python version then you'll see this kind of error.

These kinds of problems have come up in #241 and #407.

In summary, things should work if you do the following.

- Clone a fresh copy of the Ray repository.
- Make sure that `which python` finds the correct version of Python that you intend to use.
- Install Ray.

**Comment by robertnishihara** [comment]
I'm closing this for now because I think the previous comment resolves the issue. If it doesn't or if you continue to see problems, please reopen this!

**Comment by benoitpaillard** [comment]
Hi all, doing the following when upgrading from python2 to python3 does not solve the version mismatch issue 
>     * Clone a fresh copy of the Ray repository.
>     * Make sure that `which python3` finds the correct version of Python that you intend to use.
>     * Install Ray.

Both with pip3 and setup from source with pip3 install -e . --verbose

Thanks for your help
